port_ocean/core/integrations/mixins/utils.py

```diff
@@ -101,13 +140,87 @@ def is_resource_supported(
 
+async def get_items_to_parse_bulks(raw_data: dict[Any, Any], data_path: str, items_to_parse: str, items_to_parse_name: str, base_jq: str) -> AsyncGenerator[list[dict[str, Any]], None]:
+    if not data_path:
+        raw_data_serialized = json.dumps(raw_data)
+        data_path = f"/tmp/ocean/input_{uuid.uuid4()}.json"
+        with open(data_path, "w") as f:
+            f.write(raw_data_serialized)
+    output_path = f"/tmp/ocean/parsed_{uuid.uuid4()}.json"
+    items_to_parse = items_to_parse.replace(base_jq, ".")
+    delete_target = items_to_parse.split('|', 1)[0].strip() if not items_to_parse.startswith('map(') else '.'
+    base_jq_object_string = await _build_base_jq_object_string(raw_data, base_jq, delete_target)
+    jq_cmd = f"""/bin/jq '. as $all
+      | ($all | {items_to_parse}) as $items
+      | $items
+      | map({{{items_to_parse_name}: ., {base_jq_object_string}}})
+    ' {data_path} > {output_path}"""
+    try:
+        result = subprocess.run(jq_cmd, capture_output=True, text=True, shell=True, check=True)
+        if result.stderr:
+            logger.error(f"Failed to parse items for JQ expression {items_to_parse}, error: {result.stderr}")
+            yield []
+        else:
+            with open(output_path, "r") as f:
+                events_stream =get_events_as_a_stream(f, 'item', ocean.config.yield_items_to_parse_batch_size)
+                for items_bulk in events_stream:
+                    yield items_bulk
+    except Exception as e:
+        logger.error(f"Failed to parse items for JQ expression {items_to_parse}, error: {e}")
+        yield []
+    finally:
+        try:
+            if os.path.exists(output_path):
+                os.remove(output_path)
+            if os.path.exists(data_path):
+                os.remove(data_path)
+        except OSError as e:
+            logger.warning(f"Failed to cleanup temporary files: {e}")
 
 def unsupported_kind_response(
     kind: str, available_resync_kinds: list[str]
 ) -> tuple[RESYNC_RESULT, list[Exception]]:
     logger.error(f"Kind {kind} is not supported in this integration")
     return [], [KindNotImplementedException(kind, available_resync_kinds)]
 
+async def _build_base_jq_object_string(raw_data: dict[Any, Any], base_jq: str, delete_target: str) -> str:
+    base_jq_object_before_parsing = await cast(JQEntityProcessor, ocean.app.integration.entity_processor)._search(raw_data, f"{base_jq} = {json.dumps("__all")}")
+    base_jq_object_before_parsing_serialized = json.dumps(base_jq_object_before_parsing)
+    base_jq_object_before_parsing_serialized = base_jq_object_before_parsing_serialized[1:-1] if len(base_jq_object_before_parsing_serialized) >= 2 else base_jq_object_before_parsing_serialized
+    base_jq_object_before_parsing_serialized = base_jq_object_before_parsing_serialized.replace("\"__all\"", f"(($all | del({delete_target})) // {{}})")
+    return base_jq_object_before_parsing_serialized
```

Comment:
In terms of not crashing by getting empty data / null result somewhere in this function and then trying to manipulate it, are we safe? (if it's covered somewhere else in the flow please ignore, just saw some Any types so I wondered)